### ISSUE TITLE
BundleWithAssign trait

### ISSUE THREAD

**Issue by xueweiwujxw** [SpinalHDL/SpinalHDL#1071]
I always use `<<` or `>>` to connect the bus. However, I must to implement these functions in my custom bundles.
I tried to design a trait.
```scala
trait BundleWithAssignTrait[T <: BundleWithAssignTrait[T]] {
  def connectFrom(that: T): T
  def <<(that: T): T = connectFrom(that)
  def >>(into: T): T = {
    into << this.asInstanceOf[T]
    into
  }
}
```
It worked for me, and I want to open a PR for this trait.

**Comment by xueweiwujxw** [comment]
```scala
object BundleWithAssignTraitTest extends App {
  case class TestBundle()
      extends Bundle
      with IMasterSlave
      with BundleWithAssignTrait[TestBundle] {
    val valid = Bool
    val ready = Bool

    def asMaster(): Unit = {
      inWithNull(ready)
      outWithNull(valid)
    }

    def connectFrom(that: TestBundle): TestBundle = {
      that.ready := this.ready
      this.valid := that.valid
      that
    }
  }

  case class TestModule() extends Component {
    val io = new Bundle {
      val s = slave(TestBundle())
      val m = master(TestBundle())
    }

    val save = Reg(TestBundle())
    io.s >> save
    io.m << save
  }

  SpinalConfig().generateVerilog(TestModule())
}
```

**Comment by Dolu1990** [comment]
Hi
That's a good idea.
Hmm about the BundleWithAssignTrait name, maybe more something like :
IConnectable ?

**Comment by xueweiwujxw** [comment]
> Hi That's a good idea. Hmm about the BundleWithAssignTrait name, maybe more something like : IConnectable ?

It sounds great. I will push a new commit.

**Comment by xueweiwujxw** [comment]
I forgot to create a new branch.
I closed the pr #1072 and opened a new pr #1073
